bindgen/templates/NamespaceLibraryTemplate.go

```diff
@@ -3,4 +3,30 @@ func init() {
         {% for fn in initialization_fns -%}
         {{ fn }}();
         {% endfor -%}
+
+	uniffiCheckChecksums()
+}
+
+
+func uniffiCheckChecksums() {
+	// Get the bindings contract version from our ComponentInterface
+	bindingsContractVersion := {{ ci.uniffi_contract_version() }}
+	// Get the scaffolding contract version by calling the into the dylib
+	scaffoldingContractVersion := rustCall(func(uniffiStatus *C.RustCallStatus) C.uint32_t {
+		return C.{{ ci.ffi_uniffi_contract_version().name() }}(uniffiStatus)
+	})
+	if bindingsContractVersion != int(scaffoldingContractVersion) {
+		panic("UniFFI contract version mismatch: try cleaning and rebuilding your project")
```

Comment:
Nit: Message may be misleading in production apps when dynamic libraries are used.

E. g. Application binary (Go) gets updated but for some update error library (Rust) is not. User would see this message when starting the app while they had nothing to do with building the project.

Reply:
true, any suggestions what would be better? maybe I could just leave out the "help text"

Reply:
Help text may be useful for developers while troubleshooting. I think it may stay in comment line above. As for users and developers, namespace may be useful.
I think something similar to
```
panic("{{ ci.namespace() }}: UniFFI contract version mismatch")
```
would be good

Reply:
fixed